src/hoverifier.ts

```diff
@@ -82,6 +82,11 @@ export interface HoverifierOptions<C extends object, D, A> {
      * Called to get the actions to display in the hover.
      */
     getActions: ActionsProvider<C, A>
+
+    /**
+     * Whether or not hover tooltips can be pinned.
+     */
+    pinningEnabled: boolean
```

Comment:
Consider that a hoverifier is a single instance per page, and the setting for this can change during the life time of it

Reply:
The global/org/user setting `singleClickGoToDefinition`? I think it's fine to require a page reload to pick up the setting (the same applies to all settings changes).